README.md

```diff
@@ -172,7 +172,101 @@ when they expire.
+credentials:
+
+ 1) Construct an `OAuth2Credentials` instance. The callback URI should
+    be served by your app in order to complete the authorization:
+    ```python
+    >>> from xero.auth import OAuth2Credentials
+    >>>
+    >>> credentials = OAuth2Credentials(client_id, client_secret,
+    >>>                                 callback_uri=callback_uri, scope=scope)
+    ...
+    ```
+    `scope` should be the list of scopes used by the API. By default this is
+    `['openid', 'profile', 'email']`.
+
+ 2) Get the state and authentication URL, then redirect the user to it:
+
+    ```python
+    >>> from django.http import HttpResponseRedirect
+    >>> from django.core.cache import caches
+    >>>
+    >>> def start_xero_auth_view(request):
+    >>>     authorization_url = credentials.generate_url()
+    >>>     # Store the credentials state somewhere, e.g. a cache
+    >>>     cred_state = credentials.state
+    >>>     caches['mycache'].set('xero_creds', cred_state)
+    >>>     return HttpResponseRedirect(authorization_url)
+    ...
+    ```
+    `credentials.auth_state` should be stored by your app (e.g. in the database
+    or cache) as it will be used to complete the authorization during the callback.
+    The simplest way is to dump the whole credentials object by storing `credentials.state`
+
+ 3) After authorization the user will be redirected to the callback URI
+    provided (e.g. `https://example.com/oauth/xero/callback`) along with a querystring
+    containing the authentication secret. Pass the full URI including querystring to
+    `verify()`, using the credentials constructed earlier:
+    ```python
+    >>> def complete_xero_auth_view(request):
+    >>>     # Create the credentials object from wherever you stored the state
+    >>>     credentials = OAuth2Credentials(**cred_state)
+    >>>     # Then pass in the full URI the user was directed back to, including
+    >>>     # querystring
+    >>>     auth_secret = request.get_raw_uri()
+    >>>     credentials.verify(auth_secret)
+    ```
+    A token will be fetched from Xero and saved as `credentials.token`. If the credentials
+    object needs to be created again either dump the whole object using:
+    ```python
+    >>> cred_state = credentials.state
+    >>> ...
+    >>> new_creds = OAuth2Credentials(**cred_state)
+    ```
+    or just use the client_id, client_secret and token:
+    ```python
+    >>> token = credentials.token
+    >>> ...
+    >>> new_creds = OAuth2Credentials(client_id, client_secret, token=token)
+    ```
+ 4) Now the credentials may be used to authorize a Xero session. As OAuth2 allows
+    authentication for mulitple Xero Organisations, it is necessary to set
+    the tenant_id against which the xero client's queries will run.
+    ```python
+    >>> from xero import Xero
+    >>> # Use the first xero organisation (tenant) permitted
+    >>> credentials.set_default_tenant()
```

Comment:
This failed for me; `get_tenants()` returned an empty list. This is almost certainly something *I* have done wrong, but it's not clear to me what (which is, at the very least, an omission in documentation). 

It's also an uncaught error case, because the failure mode is a "list index out of range", rather than something more helpful like "no tenants available".

Reply:
Ok, I think the fix you outlined below for IndexError rather than KeyError will address this

Reply:
It will address the *error*, but won't address why I got an empty tenant list. Why did that happen? Was there something I needed to do when setting up my organization? I couldn't spot anything obvious when I set it up, and there's no mention of anything in the docs provided in this patch.

Reply:
Ah, I understand what you're saying. Probably what happened is the (default) scopes you requested didn't require access to any organisation's data, so your app could not access any tenants.

I will make 3 changes to improve this:

-  improve the documentation on setting scopes
-  improve the error message raised if tenant_id is not set
-  change the default scopes to include common use cases: read access to transactions and   contacts, and offline access in order to refresh tokens (I copied the default scope from the other PR without analysing it).

This raises a wider point about whether a decorated method on an instance of `BaseManager` would ever be called without access to any organisations (e.g. in the case of Single Sign On). From what I can see all objects in `Xero.OBJECT_LIST` are organisation specific, so I am assuming that `BaseManager` is correct to raise an exception if `self.credentials.tenant_id` is not set.

Reply:
This has been addressed in the latest commit, please review.